pkg/openstack/loadbalancer.go

```diff
@@ -1132,10 +1133,25 @@ func (lbaas *LbaasV2) getServiceAddress(clusterName string, service *corev1.Serv
 func (lbaas *LbaasV2) ensureOctaviaHealthMonitor(lbID string, name string, pool *v2pools.Pool, port corev1.ServicePort, svcConf *serviceConfig) error {
 	monitorID := pool.MonitorID
 
+	if monitorID != "" {
+		monitor, err := openstackutil.GetHealthMonitor(lbaas.lb, monitorID)
+		if err != nil {
+			return err
+		}
+		//Recreate health monitor with correct protocol if externalTrafficPolicy was changed
+		if (svcConf.healthCheckNodePort > 0 && monitor.Type != "HTTP") ||
```

Comment:
@databus23 I do not understand this change. So if externalTrafficPolicy is Local - how we can assume that all loadbalancers behind that is HTTP? This change breaks our production now, we have already 6 clusters that has this problem

monitor before this PR

```
+---------------------+--------------------------------------+
| Field               | Value                                |
+---------------------+--------------------------------------+
| project_id          | fc13c0fbb1504a41ba46bc964e787b83     |
| name                |                                      |
| admin_state_up      | True                                 |
| pools               | 5360a326-46c1-4623-bc08-4c5246276f5c |
| created_at          | 2021-01-22T09:00:34                  |
| provisioning_status | ACTIVE                               |
| updated_at          | 2021-12-15T09:54:32                  |
| delay               | 60                                   |
| expected_codes      | None                                 |
| max_retries         | 3                                    |
| http_method         | None                                 |
| timeout             | 30                                   |
| max_retries_down    | 3                                    |
| url_path            | None                                 |
| type                | TCP                                  |
| id                  | f9cc5df6-aae8-4288-9f74-fbabc08b72a5 |
| operating_status    | ONLINE                               |
| http_version        | None                                 |
| domain_name         | None                                 |
+---------------------+--------------------------------------+
```

monitor now
```
+---------------------+----------------------------------------------------------------------------------+
| Field               | Value                                                                            |
+---------------------+----------------------------------------------------------------------------------+
| project_id          | b7040dbb7f6e4a508bbcf020bb086102                                                 |
| name                | monitor_1_kube_service_kubernetes_ingress-nginx-internal_ingress-nginx-internal) |
| admin_state_up      | True                                                                             |
| pools               | fc2c0d69-5f93-45a7-b273-fed88d6c61f8                                             |
| created_at          | 2022-01-19T07:48:01                                                              |
| provisioning_status | ACTIVE                                                                           |
| updated_at          | 2022-01-20T07:25:15                                                              |
| delay               | 60                                                                               |
| expected_codes      | 200                                                                              |
| max_retries         | 3                                                                                |
| http_method         | GET                                                                              |
| timeout             | 30                                                                               |
| max_retries_down    | 3                                                                                |
| url_path            | /                                                                                |
| type                | HTTP                                                                             |
| id                  | 78ca7220-97fd-4d67-bb36-a7fc02fb79a4                                             |
| operating_status    | ONLINE                                                                           |
| http_version        | None                                                                             |
| domain_name         | None                                                                             |
+---------------------+----------------------------------------------------------------------------------+
```

I am pretty sure that our TCP services does not answer HTTP back to loadbalancer service

Reply:
The monitor endpoint is provided by kube-proxy. its on a different port (`spec.healthCheckNodePort`) and has nothing to with the tcp port of the backend port.

Reply:
in that case this is incorrectly implemented. Now this changes port monitoring also to HTTP and does not do anything with healthCheckNodePort

Reply:
this does not set monitor_port to anything, keeps protocol_port as before (pointing to normal ports). The only change this does is it will change monitoring protocol from TCP to HTTP

Reply:
ok I think I found the issue, `monitor_port` is updated only to new resources. When this is recreated to old instances the `monitor_port` is not updated. This leads to problem that everything falls to ERROR state

Reply:
old lb created before 1.23.0 version:

```
+---------------------+--------------------------------------+
| Field               | Value                                |
+---------------------+--------------------------------------+
| address             | 100.68.0.171                         |
| admin_state_up      | True                                 |
| created_at          | 2022-01-19T15:25:01                  |
| id                  | 95335116-8b07-4d08-9d60-fc2f21a48b29 |
| name                | e2etest-espsti-1wni17                |
| operating_status    | ERROR                                |
| project_id          | 9772469513404c0bb2dbfd956d885215     |
| protocol_port       | 30878                                |
| provisioning_status | ACTIVE                               |
| subnet_id           | 15773aea-ebc4-4937-9136-ba913144cc84 |
| updated_at          | 2022-01-19T16:26:41                  |
| weight              | 1                                    |
| monitor_port        | None                                 |
| monitor_address     | None                                 |
| backup              | False                                |
+---------------------+--------------------------------------+
```

new lb created with 1.23.0:

```
+---------------------+--------------------------------------+
| Field               | Value                                |
+---------------------+--------------------------------------+
| address             | 100.68.0.171                         |
| admin_state_up      | True                                 |
| created_at          | 2022-01-19T17:36:40                  |
| id                  | 705db9ac-acf9-4bc5-a7bc-65f729eac15e |
| name                | e2etest-espsti-1wni17                |
| operating_status    | ERROR                                |
| project_id          | 9772469513404c0bb2dbfd956d885215     |
| protocol_port       | 30587                                |
| provisioning_status | ACTIVE                               |
| subnet_id           | 15773aea-ebc4-4937-9136-ba913144cc84 |
| updated_at          | 2022-01-19T17:37:42                  |
| weight              | 1                                    |
| monitor_port        | 31603                                |
| monitor_address     | None                                 |
| backup              | False                                |
+---------------------+--------------------------------------+
```

Reply:
I executed manually 

```
openstack loadbalancer member set --monitor-port 31895 9598ce91-86a0-409c-8168-53355011c11d 95335116-8b07-4d08-9d60-fc2f21a48b29
openstack loadbalancer member set --monitor-port 31895 9598ce91-86a0-409c-8168-53355011c11d e1727eb6-aca0-4499-a373-dcc5c87ebfd5
openstack loadbalancer member set --monitor-port 31895 9598ce91-86a0-409c-8168-53355011c11d 9bbaea17-c79a-44f3-8af0-321314b9db15
openstack loadbalancer member set --monitor-port 31895 9598ce91-86a0-409c-8168-53355011c11d 57b06673-ac17-4f12-b6c8-98d875951c0f
openstack loadbalancer member set --monitor-port 31895 9598ce91-86a0-409c-8168-53355011c11d 12e6bf2c-7198-4bfb-9975-a0384e43f7ed
openstack loadbalancer member set --monitor-port 31895 9598ce91-86a0-409c-8168-53355011c11d 78a287c0-8cbc-48c8-a336-5c34436b55a8
openstack loadbalancer member set --monitor-port 31895 9598ce91-86a0-409c-8168-53355011c11d 38b3a28c-ea4f-44f2-900d-e9f39dd63008
openstack loadbalancer member set --monitor-port 31895 9598ce91-86a0-409c-8168-53355011c11d 56c0bd2b-56cf-41d5-9f0a-5cdb82e87684
openstack loadbalancer member set --monitor-port 31895 9598ce91-86a0-409c-8168-53355011c11d f2b9d0ca-28b3-4609-a1ea-67f8ef37afb4
openstack loadbalancer member set --monitor-port 31895 9598ce91-86a0-409c-8168-53355011c11d eeed47c6-172a-454c-b95e-6bf3d84d4634
```
and lb started to work

Reply:
@databus23 it looks like if I just delete all cloud-provider pods once more it will update those members and everything starts working. So there is some race condition in logic maybe